File: frontend/src/types/Book.ts

// This file defines the Book interface, which represents a book object in the system.
// It includes properties such as bookId, title, author, publisher, isbn, classification, category, pageCount, and price.
export interface Book {
    bookId: number
    title: string
    author: string
    publisher: string
    isbn: string
    classification: string
    category: string
    pageCount: number
    price: number
}

File: backend/Mission11_Malone/Mission11_Malone/Data/Book.cs
using System.ComponentModel.DataAnnotations;

namespace Mission11_Malone.Data
{
    // Book class to represent the Book table in the database
    public class Book
    {
        [Key]
        public int BookId { get; set; }
        [Required]
        public required string Title { get; set; }
        [Required]
        public required string Author { get; set; }
        [Required]
        public required string Publisher { get; set; }
        [Required]
        public required string ISBN { get; set; }
        [Required]
        public required string Classification { get; set; }
        [Required]
        public required string Category { get; set; }
        [Required]
        public int PageCount { get; set; }
        [Required]
        public float Price { get; set; }
    }
}
